### PR TITLE
Include Reflect.Media Version Script in Podspec

### PR DESCRIPTION
**Pull request by muthenberg**

### CODE REVIEW

File: DataCapturing.podspec [hunk 1]
```diff
@@ -41,6 +41,8 @@ This framework can be included by your App if you are going to capture sensor da
   s.source_files = 'DataCapturing/Source/**/*{.h,.m,.swift}'
   s.resources = 'DataCapturing/Source/**/*{.xcdatamodeld,.xcdatamodel,.xcmappingmodel}'
 
+  s.script_phase = { :name => 'Set version and build', :script => "#!/bin/bash\n\n# Determine plist path\nplist=${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}\n# Determine current version\nversion=$(/usr/libexec/PlistBuddy -c 'Print CFBundleShortVersionString' \"$plist\")\n# Get date of last commit\ncommitDate=$(date -r `git log -1 --date=short --pretty=format:%ct` +%Y%m%d)\n# Get current date/time\nbuildDate=$(date +%Y%m%d%H%M)\n\n# Set display version to existing version + commitDate\n/usr/libexec/PlistBuddy -c \"Set :CFBundleShortVersionString $version-$commitDate\" \"$plist\"  # Version number\n\n# Set build number to current date/time\n/usr/libexec/PlistBuddy -c \"Set :CFBundleVersion $buildDate\" \"$plist\"  # Build number\n\necho \"Version: $version-$commitDate\"\necho \"Build:   $buildDate\"\n" }
+
```
Comment (muthenberg) [inline DataCapturing.podspec:45]: This is the actual script included from RM. All the other changes below ensure that CocoaPods (which is the substitude for Gradle) is capable of actually running and building he project outside an XCode IDE.

File: DataCapturing.podspec [hunk 2]
```diff
@@ -53,10 +55,15 @@ This framework can be included by your App if you are going to capture sensor da
   # So make sure tests are always located on the same level or below the podspec.
   s.test_spec 'Tests' do |test_spec|
     test_spec.source_files = 'DataCapturing/Tests/**/*.swift'
+    test_spec.resources = 'DataCapturing/Tests/**/*.sqlite'
```
Comment (muthenberg) [inline DataCapturing.podspec:58]: Adding some missing test resources. This should enable the tests when running from outside the IDE, which was a problem I have been fighting for years now.

File: DataCapturing.podspec [hunk 3]
```diff
@@ -53,10 +55,15 @@ This framework can be included by your App if you are going to capture sensor da
...
   end
 
-  s.test_spec 'ExampleUnitTests' do |example_unit_test_spec|
-    example_unit_test_spec.source_files = 'ExampleUnitTests/**/*.swift'
+
+  s.app_spec 'Cyface' do |app_spec|
+    app_spec.source_files = 'Example/Source/**/*.swift'
+    app_spec.test_spec 'ExampleUnitTests' do |example_unit_test_spec|
+      example_unit_test_spec.source_files = 'ExampleUnitTests/**/*.swift'
+    end
```
Comment (muthenberg) [inline DataCapturing.podspec:66]: This change makes sure, that the tests for the Example app are able to find the App as a dependency, when running outside the IDE.

File: DataCapturing/Source/Model/Migrations/CoreDataManager.swift [hunk 4]
```diff
@@ -73,24 +73,20 @@ public class CoreDataManager {
...
      - Throws: `CoreDataError.invalidModelUrl`, `CoreDataError.modelNotAvailable`
      */
-    public init(storeType: String = NSSQLiteStoreType, migrator: CoreDataMigratorProtocol = CoreDataMigrator()) throws {
+    public convenience init(storeType: String = NSSQLiteStoreType, migrator: CoreDataMigratorProtocol = CoreDataMigrator()) throws {
+        let momdName = "CyfaceModel"
+        let mom = try CoreDataManager.loadModel()
+        self.init(storeType: storeType, migrator: migrator, modelName: momdName, model: mom)
+    }
+
+    init(storeType: String = NSInMemoryStoreType, migrator: CoreDataMigratorProtocol = CoreDataMigrator(), modelName: String, model: NSManagedObjectModel) {
```
Comment (muthenberg) [inline DataCapturing/Source/Model/Migrations/CoreDataManager.swift:82]: Appearantly recreating the data model multiple times is not a good idea. This is no problem in the actual iOS Application as that creates the data model only a single time. But our tests used to create it for every single test run. These (and the next two) changes enable the creation of a data model outside the CoreDataManager and providing the same data model over and over again, which should reduce warnings and performance overhead during test runs.

File: DataCapturing/Tests/DataMigrationTest.swift [hunk 5]
```diff
@@ -407,9 +406,6 @@ class DataMigrationTest: XCTestCase {
         return container
     }
 
-    private func findModelBundle() {
-
-    }
```
Comment (muthenberg) [inline DataCapturing/Tests/DataMigrationTest.swift:412]: I have now idea why this has been left here, so it has been removed.

File: DataCapturing/Tests/DataMigrationTest.swift [hunk 6]
```diff
@@ -46,7 +46,6 @@ class DataMigrationTest: XCTestCase {
     func testLoadMappingModel() throws {
         let oldResource = "3"
         let newResource = "4"
-        let subdirectory = "CyfaceModel.momd"
```
Comment (muthenberg) [inline DataCapturing/Tests/DataMigrationTest.swift:49]: removed unused property

File: DataCapturing/Tests/FileSupportTest.swift [hunk 7]
```diff
@@ -28,6 +28,7 @@ import CoreData
  - Since: 6.0.0
  */
 class FileSupportTest: XCTestCase {
+    static let dataModel = try! CoreDataManager.loadModel()
```
Comment (muthenberg) [inline DataCapturing/Tests/FileSupportTest.swift:31]: centralized data model

File: DataCapturing/Tests/PersistenceTests.swift [hunk 8]
```diff
@@ -36,14 +36,15 @@ class PersistenceTests: XCTestCase {
     var fixture: Int64!
     /// The default mode of transportation used for tests.
     let defaultMode = "BICYCLE"
+    static let dataModel = try! CoreDataManager.loadModel()
```
Comment (muthenberg) [inline DataCapturing/Tests/PersistenceTests.swift:39]: centralized data model

File: DataCapturing/Tests/SerializationTest.swift [hunk 9]
```diff
@@ -38,6 +38,7 @@ class SerializationTest: XCTestCase {
     var fixture: Int64!
     /// A manager for handling the CoreData stack.
     var coreDataStack: CoreDataManager!
+    static let dataModel = try! CoreDataManager.loadModel()
```
Comment (muthenberg) [inline DataCapturing/Tests/SerializationTest.swift:41]: centralized data model

File: DataCapturing/Tests/ServerConnectionTest.swift [hunk 10]
```diff
@@ -34,12 +34,13 @@ class ServerConnectionTest: XCTestCase {
 
     var coreDataStack: CoreDataManager!
     var oocut: ServerConnection!
+    static let dataModel = try! CoreDataManager.loadModel()
```
Comment (muthenberg) [inline DataCapturing/Tests/ServerConnectionTest.swift:37]: centralized data model

File: Example.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -391,8 +391,8 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 3D342F912554564B00E8EFA6 /* Build configuration list for PBXNativeTarget "Example" */;
 			buildPhases = (
-				3D61276425559BF7007F6884 /* Add Credentials files */,
 				341F54C0117DB5FEC8ABF74E /* [CP] Check Pods Manifest.lock */,
+				3D61276425559BF7007F6884 /* Add Credentials files */,
```
Comment (muthenberg) [inline Example.xcodeproj/project.pbxproj:395]: auto generated. just ignore this.

File: Podfile.lock [hunk 12]
```diff
@@ -29,7 +29,7 @@ EXTERNAL SOURCES:
 SPEC CHECKSUMS:
   Alamofire: 85e8a02c69d6020a0d734f6054870d7ecb75cf18
   Charts: b1e3a1f5a1c9ba5394438ca3b91bd8c9076310af
-  DataCapturing: 409378f004a7afabf0be002d8e3a518c3842934d
+  DataCapturing: b3e88ba483e5152c406bac5558921ddbfd9a80cd
```
Comment (muthenberg) [inline Podfile.lock:32]: also auto generated. you can ignore this.

File: DataCapturing.podspec (hunk 2 above)
Comment (hb0) [reply DataCapturing.podspec:58]: Maybe add some documentation about this fix then? Or else it could disappear during next refactoring?

**Review by hb0 — APPROVED** [review]

**Pull request merged**